tests/test_bcrypt.py

```diff
@@ -494,3 +499,31 @@ def test_2a_wraparound_bug():
         )
         == b"$2a$04$R1lJ2gkNaoPGdafE.H.16.1MKHPvmKwryeulRe225LKProWYwt9Oi"
     )
+
+
+# this test spawns threads and is slow, so don't run it in many threads
+@pytest.mark.parallel_threads(1)
+def test_multithreading():
+    class User:
+        def __init__(self, pw):
+            self.salt = bcrypt.gensalt(4)
+            self.hash_ = bcrypt.hashpw(pw, self.salt)
+            self.key = bcrypt.kdf(pw, self.salt, 32, 50)
+            assert self.check(pw)
+
+        def check(self, pw):
+            return bcrypt.checkpw(pw, self.hash_)
+
+    # use UUIDs as both ID and passwords
+    num_users = 50
+    user_creator = ThreadPoolExecutor(max_workers=4)
+    pws = [uuid.uuid4().bytes for _ in range(num_users)]
+
+    futures = [user_creator.submit(User, pw) for pw in pws]
```

Comment:
I don't think we need a user class at all, just a function that returns a tuple of values.